amr-wind/equation_systems/icns/source_terms/DragForcing.H

```diff
@@ -0,0 +1,49 @@
+#ifndef DRAGFORCING_H
+#define DRAGFORCING_H
+
+#include "amr-wind/equation_systems/icns/MomentumSource.H"
+#include "amr-wind/core/SimTime.H"
+#include "amr-wind/CFDSim.H"
+
+namespace amr_wind::pde::icns {
+
+/** Adds the non-linear terms from the Kosovic Model to Momentum Equation..
+ *
+ *  \ingroup icns_src NonLinearSGS
+ *
+ *  \sa NonLinearSGS
+ */
+class DragForcing : public MomentumSource::Register<DragForcing>
+{
+public:
+    static std::string identifier() { return "DragForcing"; }
+
+    explicit DragForcing(const CFDSim& sim);
+
+    ~DragForcing() override;
+
+    void operator()(
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Box& bx,
+        const FieldState fstate,
+        const amrex::Array4<amrex::Real>& src_term) const override;
+
+private:
+    const CFDSim& m_sim;
+    const amrex::AmrCore& m_mesh;
+    const Field& m_velocity;
+    Field& m_terrainBlank;
+    Field& m_terrainDrag;
+    amrex::Gpu::DeviceVector<amrex::Real> gpu_vel_ht;
```

Comment:
from a naming consistency with other stuff. Can you rename these `gpu_` to `device_` or `d_` ?